snappi/snappicommon.py

```diff
@@ -217,6 +339,68 @@ def clone(self):
         """Creates a deep copy of the current object
         """
         return self.__deepcopy__(None)
+    
+    def _validate_required(self):
+        """Validates the required properties are set
+        """
+        if getattr(self, "_REQUIRED", None) is None:
+            return
+        for p in self._REQUIRED:
+            if p in self._properties and self._properties[p] is not None:
+                continue
+            msg = "{} is a mandatory property of {}"\
+                " and should not be set to None".format(
+                p, self.__class__
+            )
+            raise ValueError(msg)
+
+    def _validate_types(self):
+        common_data_types = [list, str, int, float, bool]
+        for property, value in self._properties.items():
+            if property not in self._TYPES:
+                continue
```

Comment:
if property does not exist in self._TYPES, isn't that an error ? it would imply a bad propertly most probably.